enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,446 @@
+>installed but are missing the subscription-manger package which carries the rhsm.conf.](https://bugzilla.redhat.com/show_bug.cgi?id=1783393)
+>They are entertaining an MCO solution to work around this problem, but disabling auto-mount, and then having a 
+>solution which adds all the subscription manager in one fell swoop, would be their preference.  In the end, providing
+>an alternative to an MCO based solution **would seem** to facilitate RHCOS's future intentions and it would seem we should
+>minimally coordinate with them to facilitate how they stage future changes  
+> 2. There has been discussion elsewhere around having a more generic notion of "global resources" that any pod can
+>mount.  There is an open question of whether this proposal should tackle that.  Would we gate this enhancement on the 
+>availability of such a feature (probably "not sure")?  Upstream feature work could be implied with such an item.  But
+>what influence such a notion could have on this proposal is TBD. 
+> 3. After 4.5 epic planning it was made clear that a more general solution around entitlements, beyond just builds,
+>is desired.  PM has the to-do to investigate, prioritize, and create/assign epics accordingly.  Investigating in the 
+>interim around build specific needs via this proposal has been deemed acceptable, but certainly a subsequent, broader
+>proposal may necessitate adjustments to this one.  This proposal at the moment is straddling the fence between speaking
+>to a generic solution vs. a build specific one (though things are trending toward the former).  But a rebranding of 
+>this proposal may occur during the review process.
+> 4. We did *NOT* provide for multiple sets of entitlement credentials in 3.x.  There have been no official, RFE style  
+>registered requirements entitlements at a namespace level.  However, we've started to get (slightly) less official
+>feedback that customers want this in a more automated sense (vs. the current manual steps).  At a minimum, and is
+>noted below when alternatives are discussed, if namespaced entitlements are added with subsequent implementation
+>after global entitlements usability is improved, it needs to be seamless, and we minimally need to agree now
+>on what the order of precedence will be (with the take on that listed below).
+> 5. There has also been a question in the past needing to shard entitlements due to API rate limiting.  To date, it
+>has been stated that is not a concern for builds.  Periodic checkpoints here could be prudent.
+> 6. There is also the general question of how entitlements are propagated for our managed service offerings.  There are
+some attempts at capturing the particulars for this below, but it needs some vetting.
+> 7. With the CSI plugin based solution articulated below, there is some question as to how that is packaged, such as a)
+>a change to the existing openshift apiserver, b) a separate admission server (possibly aggregated apiserver) utilizing
```

Comment:
removed that option in next push thanks

Reply:
@bparees I went back and revisited why I thought an openshift apiserver tweak could be done to hook our new CSI themed admission plugin an I think I found it.  I do think I would need to clarify the text I originally included, but it would be a way, outside of a separate aggregated api server etc etc, and outside of patching the k8s apiserver, to hook in our new thing.

If you look at https://github.com/openshift/openshift-apiserver/blob/master/pkg/cmd/openshift-apiserver/openshiftadmission/register.go#L39

That is pulling in the support for @smarterclayton 's annotation for using imagestreams with k8s resources (https://docs.openshift.com/container-platform/3.11/dev_guide/managing_images.html#using-is-with-k8s) that you pointed me to a couple of weeks ago.

That stuff can update deployments, *sets, pods.

I think we should add a proper description of that form of hooking in to the proposal.

Thoughts?

Reply:
that is only handling openshfit-apiserver resources.  the admission plugin is registered in both the kube and openshift apiservers.

Reply:
ah gotcha thanks @bparees  ... I need to learn where that repo with our k8s patches is

Reply:
found openshift/kubernetes and openshift/kubernetes-apiserver ... didn't see anything obvious there, but did find https://github.com/openshift/api/blob/master/kubecontrolplane/v1/types.go#L42-L44 and ` oc get kubeapiservers.operator.openshift.io cluster -o yaml ` shows 

```
    imagePolicyConfig:
      internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
```

So yeah image admission sits on both apiservers.  Enough due diligence for one morning.